applications/StructuralMechanicsApplication/custom_processes/advance_in_time_strategy_high_cycle_fatigue_process.cpp

```diff
@@ -0,0 +1,251 @@
+        r_elem.CalculateOnIntegrationPoints(REVERSION_FACTOR_RELATIVE_ERROR, rev_factor_rel_error, process_info);
+        r_elem.CalculateOnIntegrationPoints(THRESHOLD_STRESS, s_th, process_info);
+        r_elem.CalculateOnIntegrationPoints(MAX_STRESS, max_stress, process_info);        
+
+        const unsigned int number_of_ip = r_elem.GetGeometry().IntegrationPoints(r_elem.GetIntegrationMethod()).size();
+        for (unsigned int i = 0; i < number_of_ip; i++){
+            if (max_stress[i] > s_th[i]) {
+                fatigue_in_course = true;
+                acumulated_max_stress_rel_error += max_stress_rel_error[i];
+                acumulated_rev_factor_rel_error += rev_factor_rel_error[i];
+            }
+        }
+    }
+    if ((acumulated_max_stress_rel_error < 1e-4 && acumulated_rev_factor_rel_error < 1e-4 && fatigue_in_course) || (DamageIndicator && acumulated_max_stress_rel_error < 1e-3 && acumulated_rev_factor_rel_error < 1e-3 && fatigue_in_course)) {
+        rAdvancingStrategy = true;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void AdvanceInTimeStrategyHighCycleFatigueProcess::TimeIncrement(double& rIncrement) 
+{
+    auto& process_info = mrModelPart.GetProcessInfo();
+    std::vector<double>  cycles_to_failure_element;
+    std::vector<int>  local_number_of_cycles;
+    std::vector<double> period;
+    std::vector<double> s_th;
+    std::vector<double> max_stress;
+    double min_time_increment;
+    double time = process_info[TIME];
+    bool current_constraints_process_list_detected = false;
+
+    double user_avancing_time = mThisParameters["fatigue"]["advancing_strategy_time"].GetDouble();
+    double user_avancing_cycles = mThisParameters["fatigue"]["advancing_strategy_cycles"].GetDouble();
+    std::vector<std::string> constraints_list = mThisParameters["fatigue"]["constraints_process_list"].GetStringArray();
+    
+    double model_part_final_time = mThisParameters["problem_data"]["end_time"].GetDouble();
+    for (unsigned int i = 0; i < constraints_list.size(); i++){
+        for (unsigned int j = 0; j < mThisParameters["processes"]["constraints_process_list"].size(); j++){
+            std::string model_part_name = mThisParameters["processes"]["constraints_process_list"][j]["Parameters"]["model_part_name"].GetString();
+            double model_part_end_time = mThisParameters["processes"]["constraints_process_list"][j]["Parameters"]["interval"][1].GetDouble();
+            if (constraints_list[i] == model_part_name && time <= model_part_end_time && !current_constraints_process_list_detected) {
+                model_part_final_time = model_part_end_time;
+                current_constraints_process_list_detected = true;
+            }
+        }
+        if (current_constraints_process_list_detected) {
+            break;
+        }
+    }
```

Comment:
the processes from the ProjectParameters are not to be used in cpp 
=> this is a very weak mechanism as the name "constraints_process_list" can also be called sth else

Reply:
I am not sure how should I solve it. Would you agree if I duplicate the info of the "constraints_process_list" in the _fatigue_ process of the ProjectParameters or you are thinking in something else?